Source/WTF/wtf/HashMap.h

```diff
@@ -198,22 +197,50 @@ class HashMap final {
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value, MappedTakeType>::type take(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*);
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    iterator find(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*);
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    const_iterator find(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*) const;
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    bool contains(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*) const;
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    MappedPeekType inlineGet(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*) const;
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    MappedPeekType get(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*) const;
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    bool remove(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*);
+    template<typename K = KeyType>
+        requires IsSmartPtrV<K>
+    MappedTakeType take(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>*);
 
     // Overloads for non-nullable smart pointer values that take the raw reference type as the parameter.
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, iterator>::type find(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&);
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, const_iterator>::type find(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&) const;
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, bool>::type contains(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&) const;
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, MappedPeekType>::type inlineGet(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&) const;
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, MappedPeekType>::type get(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&) const;
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, bool>::type remove(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&);
-    template<typename K = KeyType> typename std::enable_if<IsSmartPtr<K>::value && !IsSmartPtr<K>::isNullable, MappedTakeType>::type take(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&);
+    template<typename K = KeyType>
+        requires (IsSmartPtrV<K> && !IsSmartPtrNullableV<K>)
+    iterator find(std::add_const_t<typename GetPtrHelper<K>::UnderlyingType>&);
```

Comment:
Same question, but for a NonNullableSmartPtr concept (e.g. `template<typename T> concept NonNullableSmartPtr = SmartPtr<T> && !IsSmartPtrNullableV<T>`).